conda_index/index/__init__.py

```diff
@@ -571,6 +453,10 @@ def extract_wrapper(args: tuple):
                     # runs in thread
                     subdir, verbose, progress, subdir_path = args
                     cache = self.cache_for_subdir(subdir)
+                    # exactly these packages (unless they are un-indexable) will
+                    # be in the output repodata
+                    if self.save_fs_state:
+                        cache.save_fs_state(subdir_path)
```

Comment:
I think we use this flag OR overriding the save_fs_state() function to be a noop

Reply:
... in fact we use this flag AND we override the "get changed packages" function to return an empty list. This preserves the list of packages that belong in the index, and prevents us from trying to update the metadata cache for any of them.